File: 1920. Build Array from Permutation/build_array_from_permutation.cpp
/*
Runtime: 26 ms, faster than 10.88% of C++ online submissions for Build Array from Permutation.
Memory Usage: 16.2 MB, less than 92.36% of C++ online submissions for Build Array from Permutation.
*/

#include <vector>

using namespace std;

vector<int> buildArray(vector<int>& nums) {
  int n = nums.size();
  vector<int> ans(n);
  for(int i=0; i<n; ++i){
      ans[i] = nums[nums[i]];
  }
  return ans;
}

int main(){
  return 0;
}

File: 1920. Build Array from Permutation/build_array_from_permutation.rb
# Runtime: 130 ms, faster than 13.25% of Ruby online submissions for Build Array from Permutation.
# Memory Usage: 210.8 MB, less than 75.90% of Ruby online submissions for Build Array from Permutation.

# @param {Integer[]} nums
# @return {Integer[]}
def build_array(nums)
  ans = []
  nums.each_with_index { |val, index| 
      ans[index] = nums[val]
  }
  ans
end


# Runtime: 68 ms, faster than 96.39% of Ruby online submissions for Build Array from Permutation.
# Memory Usage: 210.8 MB, less than 46.99% of Ruby online submissions for Build Array from Permutation.

# @param {Integer[]} nums
# @return {Integer[]}
def build_array2(nums)
  nums.map {|x| nums[x]}
end
